R/locatate_header_groups_if.R

```diff
@@ -0,0 +1,138 @@
+#' @param .hook_if expression determining whether direction is hooked.
+#' @param .hook_if_rev expression determining whether direction is reverse hooked.
+#' @param default_col_header_direction Indicates which direction is given to col headers by default. Only need if "NNW" is required, rather than "N". 
+#' @param default_row_header_direction Indicates which direction is given to row headers by default. Only need if "WNW" is required, rather than "W". 
+#' @param header_fill deals with merged cells. Fills in neighbouring cells if they have the same "local_format_id", "style" or are within "borders".
+#' @param ... filter expression that identifies headers.
+#' @export
+
+locate_groups_if <-
+  function(sheet= NULL,..., direction = "N", .groupings = groupings(fmt_alignment_indent), 
+           .hook_if = hook(any(FALSE)), .hook_if_rev = hook(any(FALSE)),
+           default_col_header_direction = "N",default_row_header_direction = "W",header_fill = "local_format_id") {
+    
+    # If data_cells are missing, locate them automatically  
+    if(is.null(attr(sheet,"data_cells"))){
+      # Set data location 
+      sheet <- 
+        sheet %>% locate_data(sheet = .,!is.na(numeric))
+      
+      # Report location of data cells 
+      corner_refs <-  get_corner_cell_refs(attr(sheet,"data_cells"))
+      min_letter <- unpivotr::cols_index[corner_refs$min_col]
+      max_letter <- unpivotr::cols_index[corner_refs$max_col]
+      
+      warning(paste0("Data cells have yet not been located in this this dataframe with `locate_data()`.",
+                     " Continuing using defaults of `locate_data()`. Data cells have been located at :",
+                     min_letter,corner_refs$min_row,":",max_letter,corner_refs$max_row))
+    }
+    
+    # Add filtering if statements 
+    if(length(rlang::quos(...))>0){
+      filter_expr <- rlang::quos(...)
+      
+      filtered_header_cells <- sheet %>% dplyr::filter(!!!filter_expr)
+    } 
+    
+    # Assign attributes to objects  
+    formats <- attr(sheet, "formats")
+    tabledata <- attr(sheet, "data_cells")
+    
+    # Add annotation variables if missing  
+    added_var_list <- list(sheet,".header_label",".direction", ".value")
+    sheet <-  added_var_list %>% purrr::reduce(add_variable_if_missing)
+    
+    # Get col groups ---------------------------------------------------------------------------------------------------
+    # Get arguments for get_col_groups 
+    
+    value_ref <-   get_corner_cell_refs(tabledata)
+    header_fill_choice <-   match.arg(arg = header_fill, choices = c("local_format_id","style","borders","none"))  
+    groupings_temp <- .groupings
+    default_col_header_direction_temp <- default_col_header_direction
+    .hook_if_temp <- .hook_if  
+    .hook_if_rev_temp <- .hook_if_rev  
+    
+    min_header_index_temp <- suppressWarnings(get_header_index(sheet$.header_label,"^col_header"))  
```

Comment:
Any warnings should be avoided in `get_header_index()` rather than suppressed here.

Reply:
Addressed in 27b0d04570c640dd43f0e64ba67d6d9f2572a434.